src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Services/SyntaxFacts/ISyntaxFacts.cs

```diff
@@ -503,6 +503,10 @@ internal interface ISyntaxFacts
 
         SyntaxList<SyntaxNode> GetContentFromDocumentationCommentTriviaSyntax(SyntaxTrivia trivia);
 
+        string GetCommentText(SyntaxTrivia trivia);
```

Comment:
i would doc the semantics of this.  i.e. that it gets the contents of hte trivia, without the prefix.  also that this takes off the prefix from the start of each line if this is multiiline.

Reply:
Adding a doc comment would certainly help, but maybe it is better to remove the removal of leading `*`.

I opened  #53088 to see, what an impact a change in behavior of `GetCommentText` would have. That PR removes the *strip of leading `*` from line starts* logic. I only needed to change two test cases after that change. The affected methods (`SyntaxTriviaExtensions.GetCommentText` and `ISyntaxFacts.GetCommentText`) are both `internal`, so we can change the behavior. In case #53088 is merged, the failing test [TestExistingBlockCommentHeaderAndTextTemplateWithLeadingAsterix](https://github.com/dotnet/roslyn/blob/e481f90cfe41ee1129e1ab8f78ed1a83489bf457/src/Analyzers/CSharp/Tests/FileHeaders/FileHeaderTests.cs#L862-L876) would be green. I added some more arguments, to the description of #53088 to underpin why the behavior should be changed.